engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/VirtualFileSystem.java

```diff
@@ -0,0 +1,212 @@
+package org.enso.compiler.test.mock;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.util.Arrays;
+import java.util.stream.Stream;
+import org.enso.filesystem.FileSystem;
+
+/**
+ * {@link FileSystem} implementation that keeps all the created files in memory. Does not access the
+ * real file system.
+ */
+final class VirtualFileSystem implements FileSystem<Path>, AutoCloseable {
+  private final java.nio.file.FileSystem fileSystem;
+  private final Path inMemoryRoot;
+
+  private VirtualFileSystem(java.nio.file.FileSystem fileSystem, Path inMemoryRoot) {
+    this.fileSystem = fileSystem;
+    this.inMemoryRoot = inMemoryRoot;
+  }
+
+  static void write(Path file, String content) throws IOException {
+    Files.writeString(file, content);
+  }
+
+  @Override
+  public void close() throws Exception {
+    fileSystem.close();
+  }
+
+  public void deleteAll() throws IOException {
+    Files.walkFileTree(
+        inMemoryRoot,
+        new SimpleFileVisitor<>() {
+          @Override
+          public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+              throws IOException {
+            Files.delete(file);
+            return FileVisitResult.CONTINUE;
+          }
+
+          @Override
+          public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+            var isRoot = Files.isSameFile(inMemoryRoot, dir);
+            if (!isRoot) {
+              Files.delete(dir);
+            }
+            return FileVisitResult.CONTINUE;
+          }
+        });
+  }
+
+  public static VirtualFileSystem create() {
```

Comment:
The (missing) Javadoc of this method could mention that this creates

> implementation that keeps all the created files in memory. Does not access the real file system.

Reply:
Docs updated in a1e9f837dd62b45af51a0125667b7151f9afb7ba